packages/buidler-core/src/internal/cli/ArgumentsParser.ts

```diff
@@ -224,7 +225,10 @@ export class ArgumentsParser {
         });
       }
 
-      parsedArguments[paramName] = definition.type.parse(paramName, value);
+      // We only parse the arguments of non-internal tasks, and those only
+      // accept CLIArgumentTypes.
+      const type = definition.type as CLIArgumentType<any>;
```

Comment:
I feel like this cast shouldn't be necessary if `paramDefinitions` had a more precise type, but I tried to understand all the involved typings and got lost :disappointed: